### ISSUE TITLE
Unexplained high disk usage

### ISSUE THREAD

**Issue by benbuckland** [Azure/AKS#385]
Hi there, I have come across a periodic issue of extremely high disk usage which essentially renders the cluster useless. The only way I have found to resolve the issue is to stop and then start the VM, restarting does not resolve the issue. I'm running the latest version of Kubernetes (1.9.6) and have two Standard B2s (2 vcpus, 4 GB memory) in the cluster.

The issue does not effect CPU or Network as you can see on image attached. I am not very familiar with Linux so not quite sure how to tackle the problem and isolate the cause.

Any helper or pointers appreciated.

<img width="943" alt="more craziness on linux box snip" src="https://user-images.githubusercontent.com/29738271/40450942-38588db2-5f31-11e8-88b7-821a3736fb94.PNG">

**Comment by calvaradocl** [comment]
We are experiencing the exact same issue in our cluster, and the Disk reads increase to 30 GB with CPU. The only way to reactivate our clusters is with a restart.

We don't have any load (it's a development environment) with images of Redis, SQL Server and MongoDB and other services, but none of them except for the login are being used.

Is this a bug or a recurrent process of kubernetes or azure?

**Comment by benbuckland** [comment]
Hey @calvaradocl, I never got to the bottom of the issue but after monitoring a bunch of statistics I noticed that as the total memory used by all containers on a particular VM got close to the VM's memory limit I saw the weird behaviour. I implemented memory limit's for each containers (causing the container to be restarted when memory limit reached) and I have not seen the issue since.

Hope this helps.
